src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt

```diff
@@ -140,38 +136,94 @@ class DataDeletionService(
 
         val includeClearedEdges = deleteType == DeleteType.Hard
 
-        val entitySetIdToEntityDataKeysMap: Map<UUID, List<EntityDataKey>> = dgm
+        val entitySetIdToEntityDataKeys: Map<UUID, Set<UUID>> = dgm
                 .getEdgesConnectedToEntities(entitySetId, entityKeyIds, includeClearedEdges)
-                .filter { edge -> edge.dst.entitySetId == entitySetId && filteringNeighborEntitySetIds.contains(edge.src.entitySetId) || edge.src.entitySetId == entitySetId && filteringNeighborEntitySetIds.contains(edge.dst.entitySetId) }
+                .filter { edge ->
+                    edge.dst.entitySetId == entitySetId && filteringNeighborEntitySetIds.contains(edge.src.entitySetId)
+                            || edge.src.entitySetId == entitySetId && filteringNeighborEntitySetIds.contains(edge.dst.entitySetId)
+                }
                 .flatMap { edge -> setOf(edge.src, edge.dst) }
                 .groupBy { it.entitySetId }
+                .mapValues { it.value.map { it.entityKeyId }.toSet() }
 
         /*
-         * 2 - delete all entities
+        * 2 - check authorization on all entities
+        */
+
+        /* Check entity set permissions */
+
+        val authorizedPropertiesByEntitySets = getAuthorizedPropertyTypesForDeleteByEntitySet(
+                setOf(entitySetId) + entitySetIdToEntityDataKeys.keys, // include original entity set too
+                deleteType,
+                Optional.empty(),
+                principals
+        )
+
+        /* Check association permissions */
+
+        // only original entity set can be association entity set
```

Comment:
why can only the original entity set be an association entity set?

Reply:
Because all the other entity sets in `entitySetIdToEntityDataKeys` are either src or dst entity set ids from edges:
```
 val entitySetIdToEntityDataKeys: Map<UUID, Set<UUID>> = dgm
        .getEdgesConnectedToEntities(entitySetId, entityKeyIds, includeClearedEdges)
        ...
        .flatMap { edge -> setOf(edge.src, edge.dst) }
        ...
```

Unless src and dst entities in edges can come from association entity sets.